src/snowflake/snowpark/mock/mock_connection.py

```diff
@@ -64,6 +71,63 @@ def _build_target_path(stage_location: str, dest_prefix: str = "") -> str:
 
 
 class MockServerConnection:
+    class TableRegistry:
+        # Table registry. TODO: move to datastore
+        def __init__(self, conn: "MockServerConnection") -> None:
+            self.table_registry = {}
+            self.conn = conn
+
+        def get_fully_qualified_name(self, name: Union[str, Iterable[str]]) -> str:
+            def uppercase_and_enquote_if_not_quoted(string):
+                if string[0] == '"' and string[-1] == '"':
+                    return string
+                return f'"{string.upper()}"'
+
+            current_schema = self.conn._get_current_parameter("schema")
+            current_database = self.conn._get_current_parameter("database")
+            if isinstance(name, str):
+                name = parse_table_name(name)
+            if len(name) == 1:
+                name = [current_schema] + name
+            if len(name) == 2:
+                name = [current_database] + name
+            return ".".join(uppercase_and_enquote_if_not_quoted(n) for n in name)
+
+        def read_table(self, name: Union[str, Iterable[str]]) -> TableEmulator:
+            name = self.get_fully_qualified_name(name)
+            if name in self.table_registry:
+                return copy(self.table_registry[name])
+            else:
+                raise SnowparkSQLException(
+                    f"Table {name} does not exist"
+                )  # TODO: match exception message
+
+        def write_table(
+            self, name: Union[str, Iterable[str]], table: TableEmulator, mode: SaveMode
+        ) -> Row:
+            name = self.get_fully_qualified_name(name)
+            table = copy(table)
+            if mode == SaveMode.APPEND:
+                if name in self.table_registry:
+                    self.table_registry[name] = pd.concat(
+                        [self.table_registry[name], table]
+                    )
+                else:
+                    self.table_registry[name] = table
+            elif mode == SaveMode.IGNORE:
+                if name not in self.table_registry:
+                    self.table_registry[name] = table
+            elif mode == SaveMode.OVERWRITE:
+                self.table_registry[name] = table
+            elif mode == SaveMode.ERROR_IF_EXISTS:
```

Comment:
should we have a `else` in the end to fall back to if all these `mode == ` miss, e.g. raising error`Unrecognized SaveMode`

Reply:
True, with the current code it will return the "successfully created" message without checking.